akka_microservices/ibex35Microservice/src/main/scala/com/tapiporla/microservices/retrievers/indices/ibex35/model/Ibex35Historic.scala

```diff
@@ -0,0 +1,57 @@
+package com.tapiporla.microservices.retrievers.indices.ibex35.model
+
+import com.github.nscala_time.time.Imports._
+import com.sksamuel.elastic4s.Hit
+import com.tapiporla.microservices.retrievers.common.model.{DataInputExtractable, ElasticDocumentInsertable}
+import com.tapiporla.microservices.retrievers.common.stats.StatsGenerator.Data
+import com.tapiporla.microservices.retrievers.indices.ibex35.dao.Ibex35ESDAO
+
+object Ibex35Historic {
+  def fromMap(map: Map[String,String]) = {
+    Ibex35Historic(
+      DateTime.parse(map(Ibex35ESDAO.date), DateTimeFormat.forPattern("dd-MM-yyyy")),
+      BigDecimal(map(Ibex35ESDAO.Historic.openingValue)),
+      BigDecimal(map(Ibex35ESDAO.Historic.closeValue)),
+      BigDecimal(map(Ibex35ESDAO.Historic.minValue)),
+      BigDecimal(map(Ibex35ESDAO.Historic.maxValue)),
+      BigDecimal(map(Ibex35ESDAO.Historic.volume))
+    )
+  }
+
+  def fromHit(t: Hit): Ibex35Historic = {
+    val map = t.sourceAsMap
+    Ibex35Historic(
+      DateTime.parse(map(Ibex35ESDAO.date).toString),
+      BigDecimal(map(Ibex35ESDAO.Historic.openingValue).toString),
+      BigDecimal(map(Ibex35ESDAO.Historic.closeValue).toString),
+      BigDecimal(map(Ibex35ESDAO.Historic.minValue).toString),
+      BigDecimal(map(Ibex35ESDAO.Historic.maxValue).toString),
+      BigDecimal(map(Ibex35ESDAO.Historic.volume).toString)
+    )
+  }
+}
+
+case class Ibex35Historic(
+                           date: DateTime,
+                           openingValue: BigDecimal,
+                           closeValue: BigDecimal,
+                           minValue: BigDecimal,
+                           maxValue: BigDecimal,
+                           volume: BigDecimal
+                         ) extends ElasticDocumentInsertable with DataInputExtractable{
+
+  override def json: String = {
```

Comment:
Is this what you mentioned regarding manual json marshalling?

Reply:
Yes I have to see a way to make a json automatically